src/home/BudgetAtAGlanceWidget.js

```diff
@@ -1,7 +1,245 @@
 import React from 'react';
+import { useSelector } from 'react-redux';
+import { ProgressBar } from 'react-bootstrap';
+import { PropTypes } from 'prop-types';
+import { INCOME_ID } from '../data';
+
+/**
+ * Shows how much spent compared to budget across all categories (except income) for current month
+ * Informs user if amount spent is "Under", "On Track", or "Over" budget
+ * Spent percentage is compared to month percentage, status separated by 10%
+ * Displays progress bar
+ *
+ * @component
+ */
+
+function BudgetAtAGlance() {
+  const date = new Date('');
+
+  const firstDay = new Date(date.getFullYear(), date.getMonth(), 1);
+
+  const lastDay = new Date(date.getFullYear(), date.getMonth() + 1, 0);
+
+  // Get all transactions that do not fall under income category and reduce
+  const amountSpent = useSelector(
+    (state) => {
+      const allTransactions = state.transactions.entities
+        .filter(
+          (transaction) => transaction.date >= firstDay && transaction.date <= lastDay,
+        )
+        .filter(
+          (transaction) => transaction.categoryId !== INCOME_ID,
+        )
+        .reduce(
+          (total, transaction) => total + transaction.amount,
+          0,
+        );
+      return parseFloat(allTransactions);
+    },
+  );
+
+  // Gets categories data, excludes income category, sums target values
+  const totalBudget = useSelector(
+    (state) => {
+      const allCategories = state.categories.entities
+        .filter(
+          (category) => category.id !== INCOME_ID,
+        )
+        .reduce(
+          (total, category) => total + category.target,
+          0,
+        );
+      return parseFloat(allCategories);
+    },
```

Comment:
Does this function really need a body? Could you just return the result of the filter & reduce directly, instead of saving it in a variable and returning that variable?